### PR TITLE
set up the project directory 

### PR DESCRIPTION
**Pull request by AmitPaswan18**

### CODE REVIEW

File: src/components/admin/AddNewPoll.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    title: "",
+    options: ["Option 1"],
+  });
+
+  const handleaddpoll = async (event) => {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const title = formData.get("title");
+    const options = formData.getAll("option");
```
Comment (yuvraj-singh-codes) [inline src/components/admin/AddNewPoll.jsx]: why you are using formdata?

File: src/components/admin/AddNewPoll.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const fetchlatestPoll = () => {
+    instance
```
Comment (yuvraj-singh-codes) [inline src/components/admin/AddNewPoll.jsx]: use camelCase

File: src/components/auth/Registrationform.jsx [hunk 3]
```diff
@@ -0,0 +1,228 @@
+import Avatar from "@mui/material/Avatar";
+import Button from "@mui/material/Button";
+import CssBaseline from "@mui/material/CssBaseline";
```
Comment (yuvraj-singh-codes) [inline src/components/auth/Registrationform.jsx]: component name should be in camelCase

File: src/components/admin/AddNewOptions.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    instance
+      .get(
+        `/add_new_option?id=${editPollTitleId}&option_text=${values.addOptions}`
+      )
+      .then((response) => {
+        if (response.status === 200) {
+          dispatch(addNewOption({ loading: false }));
+          handleClose();
+        }
+      });
```
Comment (yuvraj-singh-codes) [inline src/components/admin/AddNewOptions.jsx]: you have to call API in redux not in components

File: src/components/admin/AddNewPoll.jsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+  const handleClose = () => {
+    navigate("/dashboard");
+  };
+
+  const fetchLatestPoll = () => {
+    instance
+      .get("/list_polls")
+      .then((response) => dispatch(getAllPolls(response.data.data)));
+  };
```
Comment (yuvraj-singh-codes) [inline src/components/admin/AddNewPoll.jsx]: create a slice to fetch it

File: src/components/admin/AdminPollList.jsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    dispatch(deletePollOption({ loading: true }));
+    instance
+      .get(`/delete_poll_option?id=${deletePollId}&option_text=${optionText}`)
+      .then((response) => {
+        console.log(response);
+        if (response.status === 200) {
+          dispatch(deletePollOption({ loading: false }));
+          fetchlatestPoll();
+        }
+      });
```
Comment (yuvraj-singh-codes) [inline src/components/admin/AdminPollList.jsx]: ??in all components you call API directly. You have to use redux for API calling

File: src/components/users/UserPoll.jsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+  };
+
+  const handleVotePollOption = (votePollId, optionText) => {
+    dispatch(votestarted({ votePollId: votePollId, optionText: optionText }));
+    const polltoken = localStorage.getItem("polltoken");
+    setTimeout(() => {
+      if (!isLoading) {
+        setVoted(false);
+      }
+    }, 2000);
```
Comment (yuvraj-singh-codes) [inline src/components/users/UserPoll.jsx]: why timeout?
 Is this a good approach?
what if the API response is late?
use redux , read thunk and toolkit again

File: src/components/users/UserPoll.jsx [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+    instance
+      .get(`/do_vote?id=${votePollId}&option_text=${optionText}`, {
+        headers: {
+          access_token: polltoken,
+        },
+      })
```
Comment (yuvraj-singh-codes) [inline src/components/users/UserPoll.jsx]: remove API call from every component and use redux-thunk which comes in build with toolkit

File: src/pages/Dashboard.jsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import { jwtDecode } from "jwt-decode";
+import { useNavigate } from "react-router-dom";
+
+const Dashboard = () => {
+  const showUser = localStorage.getItem("polltoken");
+  const decoded = jwtDecode(showUser);
+  const navigate = useNavigate();
+  if (!showUser) {
+    navigate("/");
+  }
```
Comment (yuvraj-singh-codes) [inline src/pages/Dashboard.jsx]: By this logic, I can access other routes excepts if even I have not logged-In yet.
Did I ask you to implement it?

File: src/components/users/UserPoll.jsx (hunk 7 above)
Comment (AmitPaswan18) [reply src/components/users/UserPoll.jsx]: sir i use this timeout just to show the voted successfully badge for 2 seconds and only appear for a successful vote

**Pull request merged**